compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -146,13 +146,11 @@ impl fmt::Display for DiagnosticLocation {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Encodable, Decodable)]
-pub enum DiagnosticId {
-    Error(String),
-    Lint {
-        name: String,
-        /// Indicates whether this lint should show up in cargo's future breakage report.
-        has_future_breakage: bool,
-    },
+pub struct IsLint {
+    /// The lint name.
+    pub(crate) name: String,
```

Comment:
Can this use the `&Lint`? Or is there an issue with crate dependency order?

Reply:
Interesting idea. Crate dependencies shouldn't be a problem. This struct has two fields:
- `name`, which is equivalent to `Lint::lower_name()`.
- `has_future_breakage`, which is a combination of `Lint::future_incompatible`, `Lint::default_level`, and `sess.opts.unstable_opts.future_incompat_test`. Adding a `sess` argument to `Diagnostic::has_future_breakage()` is awkward.

So I think keeping `IsLint` is probably best.